lib/trace/trace_parser_loop.cpp

```diff
@@ -87,34 +79,23 @@ LastFrameLoopParser::parse_call(void)
 
     /* Restart last frame when looping is requested. */
     if (call) {
-        if (call->flags & trace::CALL_FLAG_END_FRAME) {
-            lastFrameBegin = lastFrameEnd;
-            parser->getBookmark(lastFrameEnd);
+        if (starts_new_frame) {
```

Comment:
We probably only want to store the calls only `if (loopCount)`

Reply:
With that changed this is `Reviewed-by: Gert Wollny <gert.wollny@collabora.com>`

Reply:
IIUIC, loopCount will be zero only when we reach the last iteration, and by then this code doesn't execute.

Reply:
My bad,  I thought that this code is always run, but now I see that it is only run when looping is actually requested from the command line, sorry for the noise.

Reply:
Yep, I also got to think that at first.